WebInstall/explainer_same_domain.md

```diff
@@ -0,0 +1,167 @@
+* Installing content that does not pass the installability criteria (see *[installability criteria](#installability-criteria)*).
+* Enumerate if the app/related apps are installed ([see getInstalledRelatedApps](https://github.com/WICG/get-installed-related-apps/blob/main/EXPLAINER.md)).
+
+## Use Cases
+
+### **Installing a web app from the current origin**
+The site can trigger its own installation. 
+
+The current way of doing this is with the `onbeforeinstallprompt`. This only works for browsers that support PWAs and prompting. The `navigator.install` method allows a imperative way to install web content, and works for UAs that prompt and don't prompt.
+
+```javascript
+/* tries to install the current domain */
+const installApp = async () => {
+    if ('install' in navigator === false) return; // api not supported
+    try {
+            await navigator.install();
+    } catch(err) {
+        switch(err.message){
+            case 'AbortError':
+                /* Operation was aborted*/
+                break;
+           
+        }
+    }
+};
+```
+
+![Same domain install flow](./samedomaininstall.png) 
+
+The **`navigator.install()` method can overlap with some functionality of `onbeforeinstallprompt` for same-origin installation**. When the method is called it will trigger the UA to prompt for the installation of an application. This is analogous to when the end user clicks on an affordance that the UA might have to inform the user of installing. On Edge, Chrome (desktop) and Samsung Internet (mobile), this would be then the user clicks on the 'app available' banner or related UX that appears on the omnibox of the browser. For browsers that do not implement prompting, the expected behaviour is analogous to their installation paradigm. For example, in Safari (desktop) the behaviour might be that shows a dialog to add the content to the dock as an app.
+
+On UAs that support prompting, the threshold for `navigator.install()` to resolve on same-origin installations uses the same checks that `onbeforeinstallprompt` currently has for prompting (if required by the UA). The promise doesn't resolve unless the *installability criteria* is met. *Note that the criteria defined by UAs varies and can be that there is NO criteria*.
+
+When called on the same domain, the **`install()` method will trigger/open the prompt for installation the same way that using `onbeforeinstallprompt` does right now for browser that prompts.** If the domain is not installable content, then the promise returns a `DOMException` of type 'NotSupportedError'.
+
+
+## Proposed Solution
+
+### The `navigator.install` method
+
+To install a web site/app, the site/app would use the promise-based method `navigator.install([<params>]);`. This method will:
+
+* Resolve when an installation was completed.
+    * The success value will be an object that contains:
+     	*  `url`: url of the installed target.
```

Comment:
It feels like you might want to be consistent with this return value for both same-domain and cross-domain.  If cross-domain is returning back an object, shouldn't this one do the same, with the same properties?